### PR TITLE
YARN-10210. Add a RMFailoverProxyProvider that does DNS resolution on failover

### PR DESCRIPTION
**Pull request by RogPodge**
Addresses the following issue:
https://issues.apache.org/jira/browse/YARN-10210

### CODE REVIEW

File: hadoop-yarn-project/hadoop-yarn/hadoop-yarn-common/src/main/java/org/apache/hadoop/yarn/client/DefaultNoHARMFailoverProxyProvider.java [hunk 1]
```diff
@@ -95,4 +95,4 @@ public void performFailover(T currentProxy) {
   public void close() throws IOException {
     RPC.stopProxy(proxy);
   }
-}
\ No newline at end of file
+}
```
Comment (goiri) [inline hadoop-yarn-project/hadoop-yarn/hadoop-yarn-common/src/main/java/org/apache/hadoop/yarn/client/DefaultNoHARMFailoverProxyProvider.java]: Avoid this change.

Comment (RogPodge) [reply hadoop-yarn-project/hadoop-yarn/hadoop-yarn-common/src/main/java/org/apache/hadoop/yarn/client/DefaultNoHARMFailoverProxyProvider.java]: Reverted

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   2m 46s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 2 new or modified test files.  |
||| _ trunk Compile Tests _ |
| +0 :ok: |  mvndep  |   1m 53s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |  27m 55s |  trunk passed  |
| +1 :green_heart: |  compile  |  13m 52s |  trunk passed  |
| +1 :green_heart: |  checkstyle  |   1m 50s |  trunk passed  |
| +1 :green_heart: |  mvnsite  |   2m  7s |  trunk passed  |
| +1 :green_heart: |  shadedclient  |  23m 37s |  branch has no errors when building and testing our client artifacts.  |
| +1 :green_heart: |  javadoc  |   1m 46s |  trunk passed  |
| +0 :ok: |  spotbugs  |   1m  9s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   3m 20s |  trunk passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 28s |  Maven dependency ordering for patch  |
| -1 :x: |  mvninstall  |   0m 27s |  hadoop-yarn-client in the patch failed.  |
| -1 :x: |  compile  |   3m 43s |  hadoop-yarn in the patch failed.  |
| -1 :x: |  javac  |   3m 43s |  hadoop-yarn in the patch failed.  |
| -0 :warning: |  checkstyle  |   1m 28s |  hadoop-yarn-project/hadoop-yarn: The patch generated 7 new + 4 unchanged - 0 fixed = 11 total (was 4)  |
| -1 :x: |  mvnsite  |   0m 37s |  hadoop-yarn-client in the patch failed.  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| -1 :x: |  shadedclient  |   5m 55s |  patch has errors when building and testing our client artifacts.  |
| +1 :green_heart: |  javadoc  |   1m 22s |  the patch passed  |
| -1 :x: |  findbugs  |   0m 33s |  hadoop-yarn-client in the patch failed.  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   4m 22s |  hadoop-yarn-common in the patch passed.  |
| -1 :x: |  unit  |   0m 29s |  hadoop-yarn-client in the patch failed.  |
| +1 :green_heart: |  asflicense  |   0m 36s |  The patch does not generate ASF License warnings.  |
|  |   | 102m 49s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.8 Server=19.03.8 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1915/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1915 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle |
| uname | Linux bd1d31aa3407 4.15.0-74-generic #84-Ubuntu SMP Thu Dec 19 08:06:28 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / cdcb77a |
| Default Java | 1.8.0_242 |
| mvninstall | https://builds.apache.org/job/hadoop-multibranch/job/PR-1915/1/artifact/out/patch-mvninstall-hadoop-yarn-project_hadoop-yarn_hadoop-yarn-client.txt |
| compile | https://builds.apache.org/job/hadoop-multibranch/job/PR-1915/1/artifact/out/patch-compile-hadoop-yarn-project_hadoop-yarn.txt |
| javac | https://builds.apache.org/job/hadoop-multibranch/job/PR-1915/1/artifact/out/patch-compile-hadoop-yarn-project_hadoop-yarn.txt |
| checkstyle | https://builds.apache.org/job/hadoop-multibranch/job/PR-1915/1/artifact/out/diff-checkstyle-hadoop-yarn-project_hadoop-yarn.txt |
| mvnsite | https://builds.apache.org/job/hadoop-multibranch/job/PR-1915/1/artifact/out/patch-mvnsite-hadoop-yarn-project_hadoop-yarn_hadoop-yarn-client.txt |
| findbugs | https://builds.apache.org/job/hadoop-multibranch/job/PR-1915/1/artifact/out/patch-findbugs-hadoop-yarn-project_hadoop-yarn_hadoop-yarn-client.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1915/1/artifact/out/patch-unit-hadoop-yarn-project_hadoop-yarn_hadoop-yarn-client.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-1915/1/testReport/ |
| Max. process+thread count | 293 (vs. ulimit of 5500) |
| modules | C: hadoop-yarn-project/hadoop-yarn/hadoop-yarn-common hadoop-yarn-project/hadoop-yarn/hadoop-yarn-client U: hadoop-yarn-project/hadoop-yarn |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1915/1/console |
| versions | git=2.7.4 maven=3.3.9 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 45s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 2 new or modified test files.  |
||| _ trunk Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 25s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |  19m 11s |  trunk passed  |
| +1 :green_heart: |  compile  |   8m  6s |  trunk passed  |
| +1 :green_heart: |  checkstyle  |   1m 31s |  trunk passed  |
| +1 :green_heart: |  mvnsite  |   1m 42s |  trunk passed  |
| +1 :green_heart: |  shadedclient  |  18m  8s |  branch has no errors when building and testing our client artifacts.  |
| +1 :green_heart: |  javadoc  |   1m 38s |  trunk passed  |
| +0 :ok: |  spotbugs  |   1m  2s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   2m 54s |  trunk passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 25s |  Maven dependency ordering for patch  |
| -1 :x: |  mvninstall  |   0m 23s |  hadoop-yarn-client in the patch failed.  |
| -1 :x: |  compile  |   2m 53s |  hadoop-yarn in the patch failed.  |
| -1 :x: |  javac  |   2m 53s |  hadoop-yarn in the patch failed.  |
| -0 :warning: |  checkstyle  |   1m 12s |  hadoop-yarn-project/hadoop-yarn: The patch generated 7 new + 4 unchanged - 0 fixed = 11 total (was 4)  |
| -1 :x: |  mvnsite  |   0m 26s |  hadoop-yarn-client in the patch failed.  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| -1 :x: |  shadedclient  |   5m 21s |  patch has errors when building and testing our client artifacts.  |
| +1 :green_heart: |  javadoc  |   1m  5s |  the patch passed  |
| -1 :x: |  findbugs  |   0m 26s |  hadoop-yarn-client in the patch failed.  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   4m  3s |  hadoop-yarn-common in the patch passed.  |
| -1 :x: |  unit  |   0m 26s |  hadoop-yarn-client in the patch failed.  |
| +1 :green_heart: |  asflicense  |   0m 32s |  The patch does not generate ASF License warnings.  |
|  |   |  74m 32s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.8 Server=19.03.8 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1915/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1915 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle |
| uname | Linux 5a7dd8573d07 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / cdcb77a |
| Default Java | 1.8.0_242 |
| mvninstall | https://builds.apache.org/job/hadoop-multibranch/job/PR-1915/2/artifact/out/patch-mvninstall-hadoop-yarn-project_hadoop-yarn_hadoop-yarn-client.txt |
| compile | https://builds.apache.org/job/hadoop-multibranch/job/PR-1915/2/artifact/out/patch-compile-hadoop-yarn-project_hadoop-yarn.txt |
| javac | https://builds.apache.org/job/hadoop-multibranch/job/PR-1915/2/artifact/out/patch-compile-hadoop-yarn-project_hadoop-yarn.txt |
| checkstyle | https://builds.apache.org/job/hadoop-multibranch/job/PR-1915/2/artifact/out/diff-checkstyle-hadoop-yarn-project_hadoop-yarn.txt |
| mvnsite | https://builds.apache.org/job/hadoop-multibranch/job/PR-1915/2/artifact/out/patch-mvnsite-hadoop-yarn-project_hadoop-yarn_hadoop-yarn-client.txt |
| findbugs | https://builds.apache.org/job/hadoop-multibranch/job/PR-1915/2/artifact/out/patch-findbugs-hadoop-yarn-project_hadoop-yarn_hadoop-yarn-client.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1915/2/artifact/out/patch-unit-hadoop-yarn-project_hadoop-yarn_hadoop-yarn-client.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-1915/2/testReport/ |
| Max. process+thread count | 414 (vs. ulimit of 5500) |
| modules | C: hadoop-yarn-project/hadoop-yarn/hadoop-yarn-common hadoop-yarn-project/hadoop-yarn/hadoop-yarn-client U: hadoop-yarn-project/hadoop-yarn |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1915/2/console |
| versions | git=2.7.4 maven=3.3.9 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

File: hadoop-yarn-project/hadoop-yarn/hadoop-yarn-client/src/test/java/org/apache/hadoop/yarn/client/TestNoHaRMFailoverProxyProvider.java [hunk 2]
```diff
@@ -0,0 +1,272 @@
...
+  /**
+   * Test that the {@link DefaultNoHARMFailoverProxyProvider}
+   * will generate different proxies after RM IP changed
+   * and {@link DefaultNoHARMFailoverProxyProvider#performFailover(Object)}
+   * get called.
+   */
+  @Test
+  public void testDefaultFPPGetOneProxy() throws Exception {
+    // Create a proxy and mock a RMProxy
+    Proxy mockProxy1 = new TestProxy((proxy, method, args) -> null);
```
Comment (goiri) [inline hadoop-yarn-project/hadoop-yarn/hadoop-yarn-client/src/test/java/org/apache/hadoop/yarn/client/TestNoHaRMFailoverProxyProvider.java]: It looks likes Yetus is complaining about this:
error: Proxy(InvocationHandler) has protected

File: hadoop-yarn-project/hadoop-yarn/hadoop-yarn-client/src/test/java/org/apache/hadoop/yarn/client/TestRMFailoverProxyProvider.java [hunk 3]
```diff
@@ -0,0 +1,306 @@
...
+   * and {@link ConfiguredRMFailoverProxyProvider#performFailover(Object)}
+   * gets called.
+   */
+  @Test
+  public void testFailoverChange() throws Exception {
+    //Adjusting the YARN Conf
+    conf.set(YarnConfiguration.RM_HA_IDS, "rm0, rm1");
+
+    // Create two proxies and mock a RMProxy
+    Proxy mockProxy2 = new TestProxy((proxy, method, args) -> null);
```
Comment (goiri) [inline hadoop-yarn-project/hadoop-yarn/hadoop-yarn-client/src/test/java/org/apache/hadoop/yarn/client/TestRMFailoverProxyProvider.java]: Yetus complains here:
error: Proxy(InvocationHandler) has protected

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m 19s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  1s |  No case conflicting files found.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 2 new or modified test files.  |
||| _ trunk Compile Tests _ |
| +0 :ok: |  mvndep  |   1m 14s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |  24m 36s |  trunk passed  |
| +1 :green_heart: |  compile  |   9m  2s |  trunk passed  |
| +1 :green_heart: |  checkstyle  |   1m 36s |  trunk passed  |
| +1 :green_heart: |  mvnsite  |   1m 33s |  trunk passed  |
| +1 :green_heart: |  shadedclient  |  19m 26s |  branch has no errors when building and testing our client artifacts.  |
| +1 :green_heart: |  javadoc  |   1m 23s |  trunk passed  |
| +0 :ok: |  spotbugs  |   0m 55s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   2m 51s |  trunk passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 24s |  Maven dependency ordering for patch  |
| -1 :x: |  mvninstall  |   0m 21s |  hadoop-yarn-client in the patch failed.  |
| -1 :x: |  compile  |   3m 13s |  hadoop-yarn in the patch failed.  |
| -1 :x: |  javac  |   3m 13s |  hadoop-yarn in the patch failed.  |
| -0 :warning: |  checkstyle  |   1m 17s |  hadoop-yarn-project/hadoop-yarn: The patch generated 5 new + 4 unchanged - 0 fixed = 9 total (was 4)  |
| -1 :x: |  mvnsite  |   0m 24s |  hadoop-yarn-client in the patch failed.  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| -1 :x: |  shadedclient  |   4m 50s |  patch has errors when building and testing our client artifacts.  |
| +1 :green_heart: |  javadoc  |   1m  8s |  the patch passed  |
| -1 :x: |  findbugs  |   0m 26s |  hadoop-yarn-client in the patch failed.  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   4m 16s |  hadoop-yarn-common in the patch passed.  |
| -1 :x: |  unit  |   0m 54s |  hadoop-yarn-client in the patch failed.  |
| +1 :green_heart: |  asflicense  |   0m 47s |  The patch does not generate ASF License warnings.  |
|  |   |  84m 18s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.8 Server=19.03.8 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1915/3/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1915 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle |
| uname | Linux 71156e748cdd 4.15.0-74-generic #84-Ubuntu SMP Thu Dec 19 08:06:28 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / cdcb77a |
| Default Java | 1.8.0_242 |
| mvninstall | https://builds.apache.org/job/hadoop-multibranch/job/PR-1915/3/artifact/out/patch-mvninstall-hadoop-yarn-project_hadoop-yarn_hadoop-yarn-client.txt |
| compile | https://builds.apache.org/job/hadoop-multibranch/job/PR-1915/3/artifact/out/patch-compile-hadoop-yarn-project_hadoop-yarn.txt |
| javac | https://builds.apache.org/job/hadoop-multibranch/job/PR-1915/3/artifact/out/patch-compile-hadoop-yarn-project_hadoop-yarn.txt |
| checkstyle | https://builds.apache.org/job/hadoop-multibranch/job/PR-1915/3/artifact/out/diff-checkstyle-hadoop-yarn-project_hadoop-yarn.txt |
| mvnsite | https://builds.apache.org/job/hadoop-multibranch/job/PR-1915/3/artifact/out/patch-mvnsite-hadoop-yarn-project_hadoop-yarn_hadoop-yarn-client.txt |
| findbugs | https://builds.apache.org/job/hadoop-multibranch/job/PR-1915/3/artifact/out/patch-findbugs-hadoop-yarn-project_hadoop-yarn_hadoop-yarn-client.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1915/3/artifact/out/patch-unit-hadoop-yarn-project_hadoop-yarn_hadoop-yarn-client.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-1915/3/testReport/ |
| Max. process+thread count | 308 (vs. ulimit of 5500) |
| modules | C: hadoop-yarn-project/hadoop-yarn/hadoop-yarn-common hadoop-yarn-project/hadoop-yarn/hadoop-yarn-client U: hadoop-yarn-project/hadoop-yarn |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1915/3/console |
| versions | git=2.7.4 maven=3.3.9 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   2m 36s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 2 new or modified test files.  |
||| _ trunk Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 29s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |  25m 52s |  trunk passed  |
| +1 :green_heart: |  compile  |  13m  6s |  trunk passed  |
| +1 :green_heart: |  checkstyle  |   1m 57s |  trunk passed  |
| +1 :green_heart: |  mvnsite  |   1m 57s |  trunk passed  |
| +1 :green_heart: |  shadedclient  |  20m 51s |  branch has no errors when building and testing our client artifacts.  |
| +1 :green_heart: |  javadoc  |   1m 37s |  trunk passed  |
| +0 :ok: |  spotbugs  |   0m 55s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   2m 43s |  trunk passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 52s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   1m  6s |  the patch passed  |
| +1 :green_heart: |  compile  |   8m  5s |  the patch passed  |
| +1 :green_heart: |  javac  |   8m  5s |  the patch passed  |
| -0 :warning: |  checkstyle  |   1m 24s |  hadoop-yarn-project/hadoop-yarn: The patch generated 3 new + 4 unchanged - 0 fixed = 7 total (was 4)  |
| +1 :green_heart: |  mvnsite  |   1m 26s |  the patch passed  |
| -1 :x: |  whitespace  |   0m  0s |  The patch has 1 line(s) that end in whitespace. Use git apply --whitespace=fix <<patch_file>>. Refer https://git-scm.com/docs/git-apply  |
| +1 :green_heart: |  shadedclient  |  15m 40s |  patch has no errors when building and testing our client artifacts.  |
| +1 :green_heart: |  javadoc  |   1m 19s |  the patch passed  |
| +1 :green_heart: |  findbugs  |   3m  1s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   3m 58s |  hadoop-yarn-common in the patch passed.  |
| -1 :x: |  unit  |  21m 11s |  hadoop-yarn-client in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 44s |  The patch does not generate ASF License warnings.  |
|  |   | 129m 46s |   |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.yarn.client.TestApplicationClientProtocolOnHA |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.8 Server=19.03.8 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1915/4/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1915 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle |
| uname | Linux d2abacc548c6 4.15.0-74-generic #84-Ubuntu SMP Thu Dec 19 08:06:28 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 0fa7bf4 |
| Default Java | 1.8.0_242 |
| checkstyle | https://builds.apache.org/job/hadoop-multibranch/job/PR-1915/4/artifact/out/diff-checkstyle-hadoop-yarn-project_hadoop-yarn.txt |
| whitespace | https://builds.apache.org/job/hadoop-multibranch/job/PR-1915/4/artifact/out/whitespace-eol.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1915/4/artifact/out/patch-unit-hadoop-yarn-project_hadoop-yarn_hadoop-yarn-client.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-1915/4/testReport/ |
| Max. process+thread count | 576 (vs. ulimit of 5500) |
| modules | C: hadoop-yarn-project/hadoop-yarn/hadoop-yarn-common hadoop-yarn-project/hadoop-yarn/hadoop-yarn-client U: hadoop-yarn-project/hadoop-yarn |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1915/4/console |
| versions | git=2.7.4 maven=3.3.9 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m 12s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  1s |  No case conflicting files found.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 2 new or modified test files.  |
||| _ trunk Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 22s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |  21m 45s |  trunk passed  |
| +1 :green_heart: |  compile  |   8m 55s |  trunk passed  |
| -0 :warning: |  checkstyle  |   0m 30s |  The patch fails to run checkstyle in hadoop-yarn  |
| +1 :green_heart: |  mvnsite  |   1m 35s |  trunk passed  |
| +1 :green_heart: |  shadedclient  |  18m 10s |  branch has no errors when building and testing our client artifacts.  |
| +1 :green_heart: |  javadoc  |   1m 19s |  trunk passed  |
| +0 :ok: |  spotbugs  |   0m 56s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   2m 46s |  trunk passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 24s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   1m 20s |  the patch passed  |
| +1 :green_heart: |  compile  |   9m  3s |  the patch passed  |
| +1 :green_heart: |  javac  |   9m  3s |  the patch passed  |
| -0 :warning: |  checkstyle  |   1m 28s |  hadoop-yarn-project/hadoop-yarn: The patch generated 7 new + 0 unchanged - 0 fixed = 7 total (was 0)  |
| +1 :green_heart: |  mvnsite  |   1m 26s |  the patch passed  |
| -1 :x: |  whitespace  |   0m  0s |  The patch has 1 line(s) that end in whitespace. Use git apply --whitespace=fix <<patch_file>>. Refer https://git-scm.com/docs/git-apply  |
| +1 :green_heart: |  shadedclient  |  15m 38s |  patch has no errors when building and testing our client artifacts.  |
| +1 :green_heart: |  javadoc  |   1m 29s |  the patch passed  |
| +1 :green_heart: |  findbugs  |   3m 23s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   4m 20s |  hadoop-yarn-common in the patch passed.  |
| +1 :green_heart: |  unit  |  26m 49s |  hadoop-yarn-client in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 43s |  The patch does not generate ASF License warnings.  |
|  |   | 124m  5s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.8 Server=19.03.8 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1915/5/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1915 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle |
| uname | Linux b3e89708dc31 4.15.0-74-generic #84-Ubuntu SMP Thu Dec 19 08:06:28 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 0fa7bf4 |
| Default Java | 1.8.0_242 |
| checkstyle | https://builds.apache.org/job/hadoop-multibranch/job/PR-1915/5/artifact/out//home/jenkins/jenkins-slave/workspace/hadoop-multibranch_PR-1915@2/out/maven-branch-checkstyle-hadoop-yarn-project_hadoop-yarn.txt |
| checkstyle | https://builds.apache.org/job/hadoop-multibranch/job/PR-1915/5/artifact/out/diff-checkstyle-hadoop-yarn-project_hadoop-yarn.txt |
| whitespace | https://builds.apache.org/job/hadoop-multibranch/job/PR-1915/5/artifact/out/whitespace-eol.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-1915/5/testReport/ |
| Max. process+thread count | 530 (vs. ulimit of 5500) |
| modules | C: hadoop-yarn-project/hadoop-yarn/hadoop-yarn-common hadoop-yarn-project/hadoop-yarn/hadoop-yarn-client U: hadoop-yarn-project/hadoop-yarn |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1915/5/console |
| versions | git=2.7.4 maven=3.3.9 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

**Comment by hadoop-yetus** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   2m 11s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 2 new or modified test files.  |
||| _ trunk Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 24s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |  22m 38s |  trunk passed  |
| +1 :green_heart: |  compile  |   9m 49s |  trunk passed  |
| +1 :green_heart: |  checkstyle  |   1m 30s |  trunk passed  |
| +1 :green_heart: |  mvnsite  |   1m 33s |  trunk passed  |
| +1 :green_heart: |  shadedclient  |  18m 59s |  branch has no errors when building and testing our client artifacts.  |
| +1 :green_heart: |  javadoc  |   1m 25s |  trunk passed  |
| +0 :ok: |  spotbugs  |   1m  4s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   3m  7s |  trunk passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 24s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   1m 16s |  the patch passed  |
| +1 :green_heart: |  compile  |   8m 30s |  the patch passed  |
| +1 :green_heart: |  javac  |   8m 30s |  the patch passed  |
| -0 :warning: |  checkstyle  |   1m 26s |  hadoop-yarn-project/hadoop-yarn: The patch generated 3 new + 4 unchanged - 0 fixed = 7 total (was 4)  |
| +1 :green_heart: |  mvnsite  |   1m 22s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  shadedclient  |  15m 50s |  patch has no errors when building and testing our client artifacts.  |
| +1 :green_heart: |  javadoc  |   1m 20s |  the patch passed  |
| +1 :green_heart: |  findbugs  |   3m  5s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   4m  0s |  hadoop-yarn-common in the patch passed.  |
| +1 :green_heart: |  unit  |  26m 26s |  hadoop-yarn-client in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 42s |  The patch does not generate ASF License warnings.  |
|  |   | 126m 35s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.8 Server=19.03.8 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1915/6/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1915 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle |
| uname | Linux f9b53802268f 4.15.0-74-generic #84-Ubuntu SMP Thu Dec 19 08:06:28 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 0fa7bf4 |
| Default Java | 1.8.0_242 |
| checkstyle | https://builds.apache.org/job/hadoop-multibranch/job/PR-1915/6/artifact/out/diff-checkstyle-hadoop-yarn-project_hadoop-yarn.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-1915/6/testReport/ |
| Max. process+thread count | 530 (vs. ulimit of 5500) |
| modules | C: hadoop-yarn-project/hadoop-yarn/hadoop-yarn-common hadoop-yarn-project/hadoop-yarn/hadoop-yarn-client U: hadoop-yarn-project/hadoop-yarn |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1915/6/console |
| versions | git=2.7.4 maven=3.3.9 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

**Pull request merged**